packages/plugin-e2e/src/models/pages/AlertRuleEditPage.ts

```diff
@@ -41,10 +42,18 @@ export class AlertRuleEditPage extends GrafanaPage {
   }
 
   async isAdvancedModeSupported() {
-    // why not check if alertingQueryAndExpressionsStepMode feature is enabled? then we'd have to update the code when the toggle is removed.
-    const count = await this.getByGrafanaSelector(
-      this.ctx.selectors.components.AlertRules.stepAdvancedModeSwitch(QUERY_AND_EXPRESSION_STEP_ID)
-    ).count();
+    const alertingQueryAndExpressionsStepMode = await isFeatureEnabled(
+      this.ctx.page,
+      'alertingQueryAndExpressionsStepMode'
+    );
+
+    if (alertingQueryAndExpressionsStepMode) {
+      await expect(this.advancedModeSwitch).toBeVisible();
+    } else {
+      await expect(this.advancedModeSwitch).not.toBeVisible();
+    }
+
+    const count = await this.advancedModeSwitch.count();
```

Comment:
maybe I am reading this wrong but shouldn't this `.count` be made only if `alertingQueryAndExpressionsStepMode`? or maybe if it's not visible this will still return a count >0

Reply:
Great find, changed it now 👍